dFusion/dFusion.rst

```diff
@@ -184,170 +182,230 @@ TODO
-        Public: state,
-        Public: tradingWelfare,
-        Public: hashBatchInfo,
-        Public: orderHashPedersen,
-        Private: priceMatrix PxP,
-        Private: volumeVector
-        Private: balances
-        Private: orders,
-        Output: newstate
-    )
-
-The snark verifies the following:
 
-- **priceVector** and **buyVolumes** hashes to **input.hashBatchInfo** (with sha)
-- **balances** hashes to **input.state** (with pedersen)
-- **orders** hashes to **input.orderHash** (with pedersen)
-
-- for each **order** in **orders**
-    - **order.buyVolume** and **order.sellVolume** have same ratio as **order.buyToken** and **order.sellToken**
-    - Verify tnhe order only has non-zero volume if the limit price is below the market price
-    - Verify the order has not more volume than specified in **order.amount**
-    - Calculate trader surplus for this order
-    - Increment total surplus according to surplus of order
-    - Increment **totalSellVolume[order.sellToken]** by **order.sellAmount**
-    - Increment **totalBuyVolume[order.BuyToken]** by **order.buyAmount**
-    - Update the balance of the order author by subtracting **order.sellVolume** from **balance[order.sellToken]**
-    - Update the balance of the order author by addint **order.buyVolume** from **balance[order.buyToken]**
-    
-- For all tokens **t**, check that **totalSellVolume[t] == totalBuyVolume[t]** (solution doesn't mint or burn tokens)
-- Check that **tradingSurplus == input.tradingSurplus**
-- For all balances, check that **balance > 0** 
-- return **newstate** by hashing all balances together (with pedersen)
+Fraud-Proofs for auction settlements
+====================================
+
+There are a variety of fraud-proofs:
+
+State Transition & Utility & Conservation of value updates 
+---------------------------------------------------------
+- Reprovide the solution as payload
+- For each order processed, verify the order from orderstream by reconstructing the stored order rolling hash of the referenced batch
+- Verify hashed volume calldata matches committed volume hash of solution
+- For each volume:
+- Find the account + buy and sell token balance of the order belonging to this volume
+- verify merkle path of buy token balance to current state root
+- add buy volume to buy token balance
+- recompute intermediate-state-root with updated buy token leaf
+- update the trader's utility and token volumes according to the trade
+- recompute intermediate-state-root with updated token volume leaf and trader's utility
+- do the same for sell token balance (but subtract sell volume)
+- Check that resulting hash is equal to newAccountRootHash of the solution.
+
+State Transition
+----------------
+- verify via a merkle proof that the last intermediate-state-root does not hold the the claimed new-balance-state-root
+
+Utility
+-------
+- verify via a merkle proof that the last intermediate-state-root does not hold the claimed utility
+
+Conservation of value
```

Comment:
The description of this seems somewhat understated. As the code is currently written, the conservation of value proof is for a specific token.